topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -1198,24 +1210,34 @@ private void updateStatus(ReconcilableTopic reconcilableTopic) {
         var oldStatus = Crds.topicOperation(kubeClient)
             .inNamespace(reconcilableTopic.kt().getMetadata().getNamespace())
             .withName(reconcilableTopic.kt().getMetadata().getName()).get().getStatus();
+
+        // set observedGeneration
+        if ((!TopicOperatorUtil.isManaged(reconcilableTopic.kt()) || TopicOperatorUtil.isPaused(reconcilableTopic.kt())) && oldStatus == null) {
+            // observedGeneration is initialized to 0 when creating a new unmanaged or paused topic
+            reconcilableTopic.kt().getStatus().setObservedGeneration(0L);
```

Comment:
Why 0? You should use the generation from the resources. That is the version you saw and reconciled.

Reply:
The observed generation is a marker that shows that the operator works and that it saw the last update to the resource. the condition describes what it did to it. So if the generation in metadata is 10, you should still set it to 10 regardless whether it is paused or unmanaged because that is the version of the resource you handled and processed.

Reply:
Indeed, thanks.